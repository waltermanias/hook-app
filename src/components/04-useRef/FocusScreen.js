import React, { useRef } from "react";
import "../02-useEffect/effect.css";

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
  };

  return (
    <>
      <h1>FocusScreen</h1>
      <input
        ref={inputRef}
        className="form-control"
        placeholder="Your name"
      ></input>
      <button className="btn btn-outline-primary mt-5" onClick={handleClick}>
        Click
      </button>
    </>
  );
};
