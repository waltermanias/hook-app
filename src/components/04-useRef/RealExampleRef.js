import React, { useState } from "react";
import { MultipleCustomHooks } from "../03-examples/multipleCustoHooks";

import "../02-useEffect/effect.css";

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  const handleOnClick = () => {
    setShow(!show);
  };

  return (
    <>
      <h1>RealExampleRef</h1>
      <hr />

      {show && <MultipleCustomHooks />}

      <button className="btn btn-outline-primary mt-7" onClick={handleOnClick}>
        Show/Hide
      </button>
    </>
  );
};
