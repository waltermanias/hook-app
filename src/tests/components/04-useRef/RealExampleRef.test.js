import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe("RealExampleReft", () => {
  it("should render component", () => {
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  it("should show the component", () => {
    const wrapper = shallow(<RealExampleRef />);
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
