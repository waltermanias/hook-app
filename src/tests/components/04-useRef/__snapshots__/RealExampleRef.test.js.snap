// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleReft should render component 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-7"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
